File: app/src/main/java/com/prey/services/PreyBootService.java
/*******************************************************************************
 * Created by Carlos Yaconi
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.services;

import android.app.Service;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;

import com.prey.PreyLogger;

public class PreyBootService extends Service {

    private final IBinder mBinder = new LocalBinder();

    public class LocalBinder extends Binder {
        PreyBootService getService() {
            return PreyBootService.this;
        }
    }

    @Override
    public void onCreate() {
        PreyLogger.d("Prey Boot Service Started!");
    }

    @Override
    public void onDestroy() {
        PreyLogger.d("Boot Service has been stopped");
    }

    @Override
    public IBinder onBind(Intent intent) {
        return mBinder;
    }

}

File: app/src/main/java/com/prey/services/kotlin/PreyBootService.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.services.kotlin

import android.app.Service
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import com.prey.kotlin.PreyLogger

class PreyBootService : Service() {
    private val mBinder: IBinder = LocalBinder()

    inner class LocalBinder : Binder() {
        val service: PreyBootService
            get() = this@PreyBootService
    }

    override fun onCreate() {
        PreyLogger.d("Prey Boot Service Started!")
    }

    override fun onDestroy() {
        PreyLogger.d("Boot Service has been stopped")
    }

    override fun onBind(intent: Intent): IBinder? {
        return mBinder
    }
}